orderBook.js

```diff
@@ -0,0 +1,43 @@
+const reconcileOrder = (existingBook, incomingOrder) =>
+  existingBook.length
+    ? checkAllTypesMatch(existingBook, incomingOrder)
+    : addToBook(existingBook, incomingOrder)
```

Comment:
You've split your flow of control out amongst several functions, that makes it much harder to read and debug. `reconcileOrder` should be responsible for flow of control, calling stand alone helper functions. Functions should do one thing and now all of these functions have to do more.